tracks/julia/exercises/raindrops/mentoring.md

```diff
@@ -52,28 +52,25 @@ bar(x) = x < 10 ? 10 : log(x)
 If you want to benchmark lots of variants of the same function, you can define them as `raindrops1`, `raindrops2`, etc. and do some simple code generation like this:
 
 ```julia
-for i in 1:8
-   print("$i: ")
-   @btime foreach($(Symbol(:raindrops, i)), 1:1000)
-end
-```
-
-```
-1:   20.100 μs (914 allocations: 42.84 KiB)
-2:   85.000 μs (4457 allocations: 233.03 KiB)
-3:   27.800 μs (1256 allocations: 53.53 KiB)
-4:   24.300 μs (914 allocations: 42.84 KiB)
-5:   21.600 μs (914 allocations: 42.84 KiB)
-6:   35.700 μs (1457 allocations: 59.81 KiB)
-7:   47.600 μs (1589 allocations: 63.94 KiB)
-8:   86.500 μs (4457 allocations: 233.03 KiB)
+julia> for i in 1:6
+          print("$i: ")
+          @btime foreach($(Symbol(:raindrops, i)), 1:1000)
+       end
+1:   19.511 μs (914 allocations: 42.84 KiB)
+2:   21.166 μs (914 allocations: 42.84 KiB)
+3:   21.845 μs (914 allocations: 42.84 KiB)
+4:   37.608 μs (1589 allocations: 63.94 KiB)
+5:   31.062 μs (1256 allocations: 53.53 KiB)
+6:   21.833 μs (980 allocations: 44.91 KiB)
+7:   47.231 μs (1589 allocations: 63.94 KiB)
+8:   77.441 μs (4457 allocations: 233.03 KiB)
```

Comment:
Should the times be normalised perhaps? The exact time is somewhat meaningless and having values relative to the fastest one would make it easier to read imo.

Reply:
I agree, though it could be understood that these are example times.   Since they are in time order, they are already "relative to the fast one".

Reply:
>  Since they are in time order

They aren't.

Reply:
Doh!

Reply:
I've tried different options, like displaying trial judgements, but most ways I can think of add more confusion than they alleviate. Changing the order to be in perfect ascending order of times will also be more confusing I think.

Reply:
If the code were changed and suggested to create different names, then the names could be used rather than a comment indicating the solution.  The results could be ordered, and I don't know why that would be confusing.  I would be confusing if you had to comment them manually, but if it can be ordered by time and the names are part of that data, then it should remain clear.

Reply:
I mean, we could set this up real nice by using a BenchmarkGroup.

Reply:
Something like:

```julia
suite = BenchmarkGroup()

suite["if-tree"]    = @benchmarkable foreach(raindrops1, 1:1000)
suite["lookup"]     = @benchmarkable foreach(raindrops2, 1:1000)
suite["switch"]     = @benchmarkable foreach(raindrops3, 1:1000)
suite["compact-v1"] = @benchmarkable foreach(raindrops4, 1:1000)
suite["compact-v2"] = @benchmarkable foreach(raindrops5, 1:1000)
suite["compact-v3"] = @benchmarkable foreach(raindrops6, 1:1000)
suite["extendable"] = @benchmarkable foreach(raindrops7, 1:1000)
suite["IOBuffer"]   = @benchmarkable foreach(raindrops8, 1:1000)

display(run(suite))
```

Reply:
I was thinking about something like this:

```
 1.000
 1.085
 1.120
 1.928
 1.592
 1.119
 2.421
 3.969
```

instead of the absolute values. But now that I actually calculated it, I'm not sure it adds any readability. Let's just leave it as it is :)